Vector.hpp

```diff
@@ -0,0 +1,246 @@
+	 */
+	~Vector()
+	{
+
+		delete(_dynamicVector);
+	}
+
+	/*@brief Adds an element to the end of the vector.
+	*
+     **/
+	void pushBack(const T& value)
+		
+	{
+		if (_size >= _capacity)
+		{
+			Realloc(_capacity + _capacity / 2);
+
+		}
+		_dynamicVector[_size] = value;
+		_size++;
+	}
+
+	/*display the Vector*/
+	void display() const
+	{
+		for (size_t i = 0; i < _size; ++i) {
+			std::cout << _dynamicVector[i] << " ";
+		}
+		std::cout << std::endl;
+	}
+
+	/*@return returns the element in index i
+	@throw  std::out_of_range if the index is out of range*/
+	T& at(int i) const
+	{
+		if (i < 0 || i >= _size)
+		{
+			throw std::out_of_range("Index is out range");
+		}
+		return _dynamicVector[i];
+	}
+
+	/*@return returns the element in index i*/
+	T& operator[](int i)
+	{
+
+		return _dynamicVector[i];
+	}
+
+	
+	/**
+	 * @brief Overloads the equality operator to compare two vectors.
+	 *
+	 * @param other- The vector to compare with this vector.
+	 * @return True if the vectors are equal, false otherwise.
+	 */
+	bool operator==(const Vector& other) const
+	{
+		if (_size != other._size)
+		{
+			return false;
+		}
+		for (size_t i = 0; i < _size; ++i) {
+			if (_dynamicVector[i] != other._dynamicVector[i]) {
+				return false;
+			}
+		}
+		return true;
+	
+	}
+		/**
+	* @brief Overload of the << operator to print the contents of a Vector.
+	* @param os The output stream to which the contents are printed.
+	* @param vec The Vector whose contents are to be printed.
+	* @return std::ostream& A reference to the output stream.
+	*/
+	 friend std::ostream& operator<<(std::ostream& os, const Vector& vec) 
+	{	
+		os << "[";
+		for (size_t i = 0; i < vec._size; ++i) 
+		{
+			os << vec._dynamicVector[i];
+			if (i != vec._size - 1) {
+				os << ", ";
+			}
+		}
+		os << "]";
+		return os;
+	 }
+
+	/*@return returns the first element in the Vector */
+	T& front()
+	{
+		if (_size > 0)
+		{
+			return _dynamicVector[0];
+		}
+	}
+
+	/*@return returns the last element in the Vector */
+	T& back()
+	{
+		if (_size > 0)
+		{
+			return _dynamicVector[_size - 1];
+		}
+
+	}
+
+	/*@brief check if the Vector is empty
+	@return retrun true if the Vecor is empty, otherwise return false*/
+	bool empty()
+	{
+		return _size == 0;
+	}
+
+	void clear()
+	{
+
+		delete(_dynamicVector);
+		_dynamicVector = nullptr;
+		_size = 0;
+		_capacity = 0;
+	}
+	/*
+	@brief Inserts an element at the specified index in the vector.
+	@param index The position where the element should be inserted. Must be between 0 and the current size (inclusive).
+ *	@param value A constant reference to the element of type T to be inserted at the specified index.
+ *	@throw std::out_of_range if the index is out of range.*/
+
+	void insert(int index, const T& value) 
+	{
+		if (index < 0 || index > _size)
+		{
+			throw std::out_of_range("Index out of range");
+		}
+		int curr_space = _capacity - _size;
+		if (curr_space == 0)
+		{
+			Realloc(_capacity + _capacity / 2);
```

Comment:
`_capacity + _capacity / 2` is repeating itself quite a lot, perhaps a better way to do it.